src/RenderManager.cpp

```diff
@@ -24,9 +24,14 @@ void RenderManager::FreeImageErrorHandler(FREE_IMAGE_FORMAT fif, const char *mes
 	printf(" ***\n\n");
 }
 
-RenderManager::RenderManager(int mapTileWidth, int mapTileHeight, int bpp, int scaleFactor) : 
+RenderManager::RenderManager(unsigned mapTileWidth, unsigned mapTileHeight, int bpp, unsigned scaleFactor) : 
```

Comment:
I understand the desire to avoid data type conversions.

I suppose by keeping it an `int`, we are allowing implementation details (i.e. the use of FreeImage) to leak through to our API. That makes our API a bit messier and inconsistent.

Another perspective is, what happens if the user gives a negative value? Does FreeImage handle that gracefully? If it does error check, would we now be presenting the user with a FreeImage error message, which may be inconsistent with our own? Does it interpret the `bpp` value in an expected way?

As negative values are indeed invalid, they should be disallowed. Similarly, if we use an unsigned type, and they provide a value that exceeds the range of positive `int`s, then we can't really feed it through to FreeImage and expect FreeImage to handle it properly. Effectively we have an implementation restriction on `bpp` values that are too large. Either way, it seems some bounds checking is required.

Though realistically, the allowed values for `bpp` are already heavily restricted (1, 2, 4, 8, 16, 24, 32), so regardless of what representation is used, problem values should already be disallowed by basic validation. Actually, I don't think FreeImage allows a bit depth of 2, but it seems to support bit depths of 48, 64, 96, 128.

I think my gut instinct here is to use an `unsigned int` in our API, put in a range check and cast to ensure consistent treatment of values, and then defer to the library for actual supported bit depths.

Reply:
Implemented. I think this is being checked in the proper place. The downside is that the freeImageBmp is created before the check is performed. So there is a good chance that freeImage will get mad and return an error before we catch.